evm_arithmetization/src/structlog.rs

```diff
@@ -0,0 +1,159 @@
+use alloy::primitives::B256;
+use alloy::providers::ext::DebugApi;
+use alloy::providers::Provider;
+use alloy::rpc::types::trace::geth::{GethDebugTracingOptions, GethDefaultTracingOptions};
+use alloy::transports::RpcError;
+use alloy::transports::Transport;
+use alloy::transports::TransportErrorKind;
+use zerostructlog::{normalize_structlog, ZeroStructLog};
+
+/// Pass `true` for the components needed.
+fn structlog_tracing_options(stack: bool, memory: bool, storage: bool) -> GethDebugTracingOptions {
+    GethDebugTracingOptions {
+        config: GethDefaultTracingOptions {
+            disable_stack: Some(!stack),
+            // needed for CREATE2
+            disable_memory: Some(!memory),
+            disable_storage: Some(!storage),
+            ..GethDefaultTracingOptions::default()
+        },
+        tracer: None,
+        ..GethDebugTracingOptions::default()
+    }
+}
+
+// Gets the struct logs with the necessary logs for debugging.
+pub async fn get_structlog_for_debug<ProviderT, TransportT>(
```

Comment:
You should probably rebase this and merge with @einar-polygon branch and PR https://github.com/0xPolygonZero/zk_evm/pull/682. There structlogs retrieval is optimized to be one call per block.